### DISCUSSION TITLE
Improvements on marker and session summary

### DISCUSSION THREAD

**Discussion by Shriukan33** [transport-nantes/tn_web#1134] [Ideas]
All screenshots presented in this post are from this figma page (read only) : [Link](https://www.figma.com/file/OPzWOPPJxWaTonVL52wvtb/Mobilito?node-id=0%3A1&t=FfNZP9gVts2yJ5s6-1) 

# Markers

I've created sets of icons that may replace the marker, with no consideration for implementation. The purpose is to see what we like and want to tend to.

![image](https://user-images.githubusercontent.com/70256364/212697208-8b767a5d-62b8-4c8d-9219-d33bc8320aa6.png)

## Pie in marker
I remember you mentioned your (@JeffAbrahamson ) interest in having the pie chart within the marker. So I produced 2 versions of this, one that is full width, the other with some padding. 

![image](https://user-images.githubusercontent.com/70256364/212697295-c4e26c2d-9cc4-408e-bb39-57d2d6e16e88.png)

![image](https://user-images.githubusercontent.com/70256364/212697341-1589c4df-abd0-4fe9-bd29-74e6b87ec0f0.png)

I find the pie chart too hard to read at this size.

## Biggest modal as a pin

I tried something easier to read and that gives only 1 info : What's the highest mode in the modal share of this session.
In the example, I've recorded a session where the public transports are the preferred mean of transportation. 

![image](https://user-images.githubusercontent.com/70256364/212698087-074e856d-3939-4489-8d09-ff26417064b6.png)

I did a set that has the mobilitains-blue color, and one that uses the other colors we have in our palette. 
The colored version would be nice to tell apart the different modes on a more global map, it could visually draw territories using color. (Pedestrian land, car land etc)

# More info on summaries 

I've added a section on top of the map that takes inspiration from Strava to add more info on the mobilito sessions.

![image](https://user-images.githubusercontent.com/70256364/212699997-09a4f7bd-c23a-417f-ad8a-9b080ef146d4.png)

From left to right :  Most observed event |  Session's duration  |  Number of events

What do you think ?

**Comment by JeffAbrahamson** [comment]
Colours have meaning.  We don't consistently identify modes by colours.  We do have a visual guide that makes other identifications with colour.  We should stick to that.  Or redesign everything from the ground up, which we have neither time nor in-house competence to do.

**Comment by JeffAbrahamson** [comment]
The markers with pie chart are really cool, but consider that they'd have to all be dynamically generated.  So I have trouble imagining we could prioritise this given the time necessary to do it even half-way well.

**Reply by JeffAbrahamson** [reply]
There are also issues about how to aggregate them when you have several observations in a small area.

**Reply by JeffAbrahamson** [reply]
Not insurmountable, but doesn't sound like something for which we have resources.

**Reply by Shriukan33** [reply]
I like them too but they are really not that big, we couldn't have labels on it. :/

Also, I share your concerns about implementation, that would probably be a bit hard to do. I didn't research it, but that's some precise front-end feature/.

**Comment by JeffAbrahamson** [comment]
I'm concerned that showing a majority mode will show cars most of the time.  That's hardly an interesting or useful message for us to show.  Thus the interest in pie charts.
![image](https://user-images.githubusercontent.com/1220683/212719016-f8457de7-90cc-4574-abef-b927ce01161d.png)

**Reply by Shriukan33** [reply]
I'm seeing it to be either pedestrian or car, depending if you're in a pedestrian area that is largely walkable (like some areas in Bordeaux barely see any car, only peds)

**Comment by JeffAbrahamson** [comment]
When I suggested we composite the pie chart on the map, I was really thinking we use PIL to grab the map, then grab the pie chart, then composite the pie chart into, say, the upper left corner.

**Reply by Shriukan33** [reply]
... That seem reasonable haha
The pie embedded in marker seem like a hassle to implement, but maybe it's just that I don't know Pillow all that well.
I'll start with the pie chart at 0,0 px like you suggested, and will explore the tool to see how we'd do the embedded pin.

**Reply by Shriukan33** [reply]
![image](https://user-images.githubusercontent.com/70256364/212884699-156c966c-91af-4947-8b93-cfb35c263534.png)

I tried using the pie chart image, but the image includes also the text and labels. 

Would it be possible to embed the labels within the pie chart and only render the pie ? Throwing ideas.

Btw I didn't need pillow, I used responsive to make the images overlap.

**Reply by Shriukan33** [reply]
![image](https://user-images.githubusercontent.com/70256364/212924872-45fdaae7-f021-451a-9151-93e305ef2310.png)

(smaller)
![image](https://user-images.githubusercontent.com/70256364/212925247-116ac4ac-c69b-4f2e-b754-bc891557b397.png)


Managed to produce something that look like this but without labels it's unreadable, also the chart uses the same color for several items, it seems
